app/views/admin/users/index.html.haml

```diff
@@ -17,7 +17,9 @@
       %tr
         %td= user.email
         %td.actions
-          = link_to "Edit", edit_admin_user_path(user), class: "btn btn-small"
-          = link_to "Delete", admin_user_path(user), method: :delete, data: { confirm: "Are you sure you want to delete #{user}?" }, class: "btn btn-small"
+          - if policy(user).edit?
```

Comment:
Just a thought...

People are smart; there is a column in that table do to something even if the buttons in it are missing.

Maybe visually more appealing if all the options are listed but you cannot use them.
![](http://puu.sh/jGJDr/d50ca9ff1d.png)

Randomly thought of analogy:
If a person is not meant to access the room behind a door, you don't hide it, you lock it. :wine_glass: :smoking: